### PR TITLE
Final app modernization recipes

### PR DESCRIPTION
**Pull request by bryanfriedman**

### CODE REVIEW

File: content/guides/spring/offload-http-sessions-spring-session.md [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+## Configure the Service to Bind to the Application
+
+By default, an [Apache Tomcat](http://tomcat.apache.org) web server instance is configured to store all sessions and associated data in memory. Under certain circumstances it is desirable to persist sessions using a repository. You can configure Tomcat to do so by binding an appropriate service.
+
+The Tanzu Application Service GO Router uses the `jsessionid` plus a `vcap_id` to establish sticky sessions. Session replication breaks with sticky sessions at the GO router which is one of the reasons Spring Session is used. By default Spring Session creates a custom `SESSION` cookie to house an application’s HTTP session. See the [documentation](https://docs.pivotal.io/pivotalcf/concepts/http-routing.html)).
+
+
+
+> Ensure your session object implements [serializable](https://docs.oracle.com/javase/tutorial/jndi/objects/serial.html).
```
Comment (Maarc) [inline content/guides/spring/offload-http-sessions-spring-session.md:24]: > Ensure your session object implements [java.io.Serializable](https://docs.oracle.com/javase/tutorial/jndi/objects/serial.html).

File: content/guides/spring/spring-cloud-connectors-datasources.md [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+
+## Cloud Foundry Connectors
+
+Spring Cloud Connectors work on a tiered model. At the bottom level, there is a *Cloud Connector* which detects the platform being used (e.g. Cloud Foundry, Heroku, etc.). Above that, you have service connectors which detect bound services and expose `ServiceInfo` objects. You can interact with the cloud connector system programmatically if you so choose, but it also slots in directly with Spring Boot.
+
+The key thing to remember about the service connectors is that they all perform dynamic detection of the service info type based on patterns detected within the *credentials* field of the user-provided service. For example, if the scheme **sqlserver://** is detected within the **jdbcUrl** field of **credentials**, then the SQL Server service connector will be used, and the appropriate plumbing will be slotted in. Refer to the documentation for a list of the ways in which detection takes place, the supported types, and how to use the connectors programmatically.
+
+As an example, here's what it might look like to create a user-provided service that can be detected as a SQL Server binding with Spring Cloud connectors:
+
+``
```
Comment (Maarc) [inline content/guides/spring/spring-cloud-connectors-datasources.md:50]:
```
Would be better here formatting-wise

File: content/guides/spring/spring-cloud-connectors-datasources.md [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+cf cups sqlDataSource1 -p '{\"jdbcUrl\":\"jdbc:sqlserver://host:port;DatabaseName=theDb;user=user;password=password;\"}'
+``
```
Comment (Maarc) [inline content/guides/spring/spring-cloud-connectors-datasources.md:52]: Here are well. Would recommend ```

File: content/guides/spring/spring-cloud-connectors.md [hunk 4]
```diff
@@ -0,0 +1,243 @@
...
+
+Spring Cloud Connectors uses the `CloudConnector` interface to provide cloud platform support. A `CloudConnector` implementation for a particular cloud platform is responsible for detecting when the application is running in that cloud platform, obtaining information about the application from the cloud platform, and obtaining information about the services that are bound to the application.
+
+## Cloud Service Support
+
+You can extend Spring Cloud Connectors to support additional services, including services that are specific to your own environment or application. Spring Cloud Connectors uses two interfaces to provide cloud service support:
+
+* `ServiceInfo` models the information required to connect to the service. In the case of a database service, a ServiceInfo implementation might include fields for host, port, database name, username, and password; in the case of a web service, it might include fields for URL and API key.
+
+*`ServiceInfoCreator` creates `ServiceInfo` objects based on the service information collected by a cloud connector. A ServiceInfoCreator implementation is specific to a cloud platform.
```
Comment (Maarc) [inline content/guides/spring/spring-cloud-connectors.md:32]: Space missing between "*" and "`ServiceInfoCreator`".

It should be:
* `ServiceInfoCreator`

File: content/guides/spring/bootifying-java-apps.md [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+
+Before changing anything in your application, you should also create a separate bootification branch within your version control system (e.g., Git). This way, you will be capable of tracking and viewing differences between the original application and the bootified one and rolling back your changes if necessary.
+
+### 4. Adjust application dependencies, packaging, and code structure
+
+Now it is time to update your application dependencies, packaging and create a Spring Boot application class.
+
+If you want to use either Maven or Gradle, [Spring Initializr](http://start.spring.io/) is the recommended approach for generating a new Spring Boot application. Spring Initializr is like a shopping cart for all of the dependencies you might need for your application. It offers a fast way to pull in all of the dependencies you need for an application and does a lot of the setup for you. Just Navigate to [Spring Initializr](http://start.spring.io/), fill in the metadata, add the dependencies relevant to your application ... and generate the new project. Make sure to carefully pick the right Spring Boot starters for your application.
```
Comment (BrianMMcClain) [inline content/guides/spring/bootifying-java-apps.md:52]:
```suggestion
If you want to use either Maven or Gradle, [Spring Initializr](http://start.spring.io/) is the recommended approach for generating a new Spring Boot application. Spring Initializr is like a shopping cart for all of the dependencies you might need for your application. It offers a fast way to pull in all of the dependencies you need for an application and does a lot of the setup for you. Just Navigate to [Spring Initializr](http://start.spring.io/), fill in the metadata, add the dependencies relevant to your application, and generate the new project. Make sure to carefully pick the right Spring Boot starters for your application.
```

File: content/guides/spring/bootifying-java-apps.md [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+
+At this stage, make sure you understand Spring Boot auto-configuration and [how to adjust it](https://docs.spring.io/spring-boot/docs/current/reference/html/using-boot-auto-configuration.html).
+
+### 5. (Optional) Configure remote deployment
+
+It typically makes sense to start working locally on your developer machine. Once you want to connect your application to external resources (e.g., database, messaging middleware, external services) or test it in a more realistic environment, deploying your Spring Boot application to a remote runtime environment or server is an excellent idea.
+
+For this reason, from this point on you can consider deploying the application on [VMware Tanzu Application Service] (https://tanzu.vmware.com/application-service) (TAS) or a remote testing server. TAS is based on the Cloud Foundry open-source project. It is built to be the best runtime for Spring and Spring Boot applications, providing a lean, integrated, and efficient user experience. If you want to deploy your application there, follow the corresponding "[getting started](https://docs.cloudfoundry.org/buildpacks/java/getting-started-deploying-apps/gsg-spring.html)" deployment guide.
```
Comment (BrianMMcClain) [inline content/guides/spring/bootifying-java-apps.md:62]:
```suggestion
For this reason, from this point on you can consider deploying the application on [VMware Tanzu Application Service](https://tanzu.vmware.com/application-service) (TAS) or a remote testing server. TAS is based on the Cloud Foundry open-source project. It is built to be the best runtime for Spring and Spring Boot applications, providing a lean, integrated, and efficient user experience. If you want to deploy your application there, follow the corresponding "[getting started](https://docs.cloudfoundry.org/buildpacks/java/getting-started-deploying-apps/gsg-spring.html)" deployment guide.
```

**Review by BrianMMcClain — CHANGES_REQUESTED** [review]

**Review by BrianMMcClain — APPROVED** [review]

**Pull request merged**